helpers/voting/resolveAncillaryData.ts

```diff
@@ -0,0 +1,183 @@
+import { RequestAddedEvent } from "@uma/contracts-frontend/dist/typechain/core/ethers/VotingV2";
+import { Contract } from "ethers";
+import {
+  BytesLike,
+  defaultAbiCoder,
+  Interface,
+  keccak256,
+} from "ethers/lib/utils";
+import { getProvider } from "helpers/config";
+import { decodeHexString } from "helpers/web3/decodeHexString";
+
+// bytes, bytes32, address do NOT have "0x" prefix
+// uint are represented as decimal string
+
+const abi = [
+  {
+    anonymous: false,
+    inputs: [
+      {
+        indexed: true,
+        internalType: "address",
+        name: "requester",
+        type: "address",
+      },
+      {
+        indexed: false,
+        internalType: "bytes32",
+        name: "identifier",
+        type: "bytes32",
+      },
+      {
+        indexed: false,
+        internalType: "uint256",
+        name: "time",
+        type: "uint256",
+      },
+      {
+        indexed: false,
+        internalType: "bytes",
+        name: "ancillaryData",
+        type: "bytes",
+      },
+      {
+        indexed: true,
+        internalType: "bytes32",
+        name: "childRequestId",
+        type: "bytes32",
+      },
+      {
+        indexed: true,
+        internalType: "bytes32",
+        name: "parentRequestId",
+        type: "bytes32",
+      },
+    ],
+    name: "PriceRequestBridged",
+    type: "event",
+  },
+];
+
+export async function resolveAncillaryDataForRequests<
+  T extends Parameters<typeof resolveAncillaryData>[0]
+>(requests: T[]): Promise<T[]> {
+  const resolvedAncillaryData = await Promise.all(
+    requests.map((request) => resolveAncillaryData(request))
+  );
+  return requests.map((request, i) => ({
+    ...request,
+    ancillaryData: resolvedAncillaryData[i],
+  }));
+}
+
+export async function resolveAncillaryData(
+  args: Pick<RequestAddedEvent["args"], "ancillaryData" | "time" | "identifier">
+): Promise<string> {
+  const decodedAncillaryData = decodeHexString(args.ancillaryData);
+  const { ancillaryDataHash, childOracle, childChainId, childBlockNumber } =
+    extractMaybeAncillaryDataFields(decodedAncillaryData);
```

Comment:
does this guaranee the ordering of fields and that no other information is present (e.g. so no one can spoof this on L1 OOv2)

Reply:
no it doesn't I'll have to add this still

Reply:
@Reinis-FRP this should do it https://github.com/UMAprotocol/voter-dapp-v2/pull/364/commits/65c2211795b3bb6bbbf045c73454c0ac8bf2700e